samples/nrf9160/nrf_cloud_rest_agps/src/main.c

```diff
@@ -0,0 +1,604 @@
+		LOG_ERR("Could not initialize modem info module, error: %d", err);
+		return err;
+	}
+
+	err = modem_info_params_init(modem_info);
+	if (err) {
+		LOG_ERR("Could not initialize modem info parameters, error: %d", err);
+		return err;
+	}
+
+	err = modem_info_params_get(modem_info);
+	if (err) {
+		LOG_ERR("Could not obtain cell information, error: %d", err);
+		return err;
+	}
+
+	return 0;
+}
+
+int init(void)
+{
+	int err;
+
+	err = init_led();
+	if (err) {
+		LOG_ERR("LED initialization failed");
+		return err;
+	}
+
+	err = nrf_modem_lib_init(NORMAL_MODE);
+	if (err) {
+		LOG_ERR("Failed to initialize modem library: %d", err);
+		LOG_ERR("This can occur after a modem FOTA.");
+		return err;
+	}
+
+	err = at_cmd_init();
+	if (err) {
+		LOG_ERR("Failed to initialize AT commands, err %d", err);
+		return err;
+	}
+
+	gps_dev = device_get_binding("NRF9160_GPS");
+	if (gps_dev != NULL) {
+		err = gps_init(gps_dev, gps_handler);
+		if (!err) {
+			LOG_INF("GPS INITD");
+		}
+	}
+
+	err = set_device_id();
+	if (err) {
+		LOG_ERR("Failed to set device ID, err %d", err);
+		return err;
+	}
+
+	LOG_INF("Device ID: %s", log_strdup(device_id));
+
+	err = dk_buttons_init(button_handler);
+	if (err) {
+		LOG_ERR("Failed to initialize button: err %d", err);
+		return err;
+	}
+
+	err = request_jitp();
+	if (err) {
+		LOG_WRN("User JITP request failed");
+		err = 0;
+	}
+
+	return 0;
+}
+
+static void lte_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		LOG_DBG("LTE_LC_EVT_NW_REG_STATUS: %d", evt->nw_reg_status);
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			break;
+		}
+
+		LOG_INF("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_DBG("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		current_cell.id = evt->cell.id;
+		current_cell.tac = evt->cell.tac;
+		break;
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+	case LTE_LC_EVT_PSM_UPDATE:
+	case LTE_LC_EVT_EDRX_UPDATE:
+	case LTE_LC_EVT_RRC_UPDATE:
+	default:
+		break;
+	}
+}
+
+static int connect_to_network(void)
+{
+	int err;
+
+	LOG_INF("Waiting for network..");
+
+	(void)k_sem_take(&lte_connected, K_NO_WAIT);
+
+	err = lte_lc_init_and_connect_async(lte_handler);
+	if (err) {
+		LOG_ERR("Failed to init modem, err %d", err);
+	} else {
+		k_sem_take(&lte_connected, K_FOREVER);
+		LOG_INF("Connected");
+		(void)set_led(1);
+	}
+
+	return err;
+}
+
+
+void main(void)
+{
+	LOG_INF("Starting \"%s\"..", log_strdup(CONFIG_REST_SAMPLE_NAME));
+
+	char rx_buf[REST_RX_BUF_SZ];
+	int agps_sz;
+	struct jwt_data jwt = {
+#if defined(CONFIG_REST_ID_SRC_INTERNAL_UUID)
+		.subject = NULL,
+#else
+		.subject = device_id,
+#endif
+		.audience = NULL,
+		.exp_delta_s = 0,
```

Comment:
we should use an expiration time (or nrf_cloud_generate_jwt)

Reply:
Sure, I can switch to your new function after I share this with the customer today.

What value do you suggest in the meantime?

Reply:
doesn't matter much, i just don't think it is good to provide examples with a JWT that works **forever**.
how about 5 minutes?